node/src/components/contract_runtime/rewards.rs

```diff
@@ -504,14 +511,29 @@ pub(crate) fn rewards_for_era(
                     .checked_mul(&rewards_info.reward(signed_block_era)?)
                     .ok_or(RewardsError::ArithmeticOverflow)?;
 
+                trace!(
+                    signer=?signing_validator,
+                    amount=%contribution_reward.to_integer(),
+                    block=%block.height,
+                    signed_block=%signed_block_height,
+                    "signature contribution reward"
+                );
+                trace!(
+                    collector=?block.proposer,
+                    signer=?signing_validator,
+                    amount=%collection_reward.to_integer(),
+                    block=%block.height,
+                    signed_block=%signed_block_height,
+                    "signature collection reward"
+                );
                 increase_value_for_key_and_era(
                     signing_validator,
                     signed_block_era,
                     contribution_reward,
                 )?;
                 increase_value_for_key_and_era(
                     block.proposer.clone(),
-                    signed_block_era,
+                    current_era_id,
```

Comment:
I am quite confused about how this fix actually works, since the data structure ``increase_value_for_key_and_era`` mutates is a bit of an accounting fiction that just ends up being rolled up into a PublicKey -> Vec<U512>. I.e., we already _found_ the validator key for the recently joined validator before we even reach this bit of the code, no?

Reply:
Yes, but this is not the only place we look for the validator. After this is completed, the auction contract will attempt to distribute the rewards and it will do a lookup on the validators as well in order to figure out how much goes to the delegators. It is at this point when the process fails: since without the fix, the validator in question will have a nonzero total reward calculated for the previous era, the auction contract will attempt to find the validator in the snapshot for the previous era, but fail and exit with an error.

Reply:
Ah, I see, thank you